utils/githooks/find_base.sh

```diff
@@ -19,34 +19,24 @@ fi
 
 # Try and use the gh command to work out the target branch, or if not installed
 # then assume origin/master.
-if command -v gh > /dev/null 2>&1; then
+if ${USE_GH:-true} && command -v gh > /dev/null 2>&1; then
     # If there is no PR created yet then do not check anything.
     if ! TARGET="$ORIGIN"/$(gh pr view "$BRANCH" --json baseRefName -t "{{.baseRefName}}"); then
-        TARGET=HEAD
+        TARGET=""
     else
         state=$(gh pr view "$BRANCH" --json state -t "{{.state}}")
         if [ ! "$state" = "OPEN" ]; then
-            TARGET=HEAD
+            TARGET=""
         fi
     fi
-else
-    # With no 'gh' command installed, use the "closest" branch as the target,
-    # calculated as the sum of the commits this branch is ahead and behind.
+fi
+
+if [ -z "$TARGET" ]; then
+    # With no 'gh' command installed, or no PR open yet, use the "closest" branch
+    # as the target, calculated as the sum of the commits this branch is ahead and
+    # behind.
     # check master, then current release branches, then current feature branches.
-    # shellcheck disable=SC2034
-    all_bases=("master" "release/2.4" "feature/cat_recovery" "feature/multiprovider")
-    TARGET="$ORIGIN/master"
-    min_diff=-1
-    for base in "${all_bases[@]}"; do
-        git rev-parse --verify "$ORIGIN/$base" &> /dev/null || continue
-        commits_ahead=$(git log --oneline "$ORIGIN/$base..HEAD" | wc -l)
-        commits_behind=$(git log --oneline "HEAD..$ORIGIN/$base" | wc -l)
-        commits_diff=$((commits_ahead + commits_behind))
-        if [ "$min_diff" -eq -1 ] || [ "$min_diff" -gt "$commits_diff" ]; then
-            TARGET="$ORIGIN/$base"
-            min_diff=$commits_diff
-        fi
-    done
+    TARGET="$ORIGIN/$(utils/rpms/packaging/get_release_branch "feature/cat_recovery feature/multiprovider")"
```

Comment:
I'm almost tempted to say we don't need to look at the feature branches, since as you pointed out before, they should be up-to-date with master anyway.

But I think we could handle that later after consulting with other relevant parties

Reply:
It is.  We actually have lots of that going on because it's a necessary evil we need to do to minimize the amount of code we add to `Jenkinsfile`'s `pipeline {}` block due to the design flaw of how Jenkins renders that into Java.

Reply:
From my limited understanding of the issue, it's a limit on the _**static**_ pipeline size. But function calls should by dynamic. So I would think something like this should work, even putting groovy scripts in the `daos` repo
```
load("my_func.groovy").foo()
```

Anyway, that's a much larger problem to address so doesn't need to block this PR.